src/Features/Core/Portable/Diagnostics/HostAnalyzerManager.cs

```diff
@@ -154,6 +160,17 @@ private ImmutableArray<DiagnosticDescriptor> GetDiagnosticDescriptorsCore(Diagno
             return descriptors;
         }
 
+        /// <summary>
+        /// Returns cached type name for the given <paramref name="analyzer"/>.
+        /// </summary>
+        public string GetAnalyzerTypeName(DiagnosticAnalyzer analyzer)
+        {
+            return _analyzerTypeNameCache.GetValue(analyzer, ComputeAnalyzerTypeName);
+
+            static string ComputeAnalyzerTypeName(DiagnosticAnalyzer analyzer)
```

Comment:
does a static local function get cached?  i'm not sure it does.  consider just an actual lambda, or a delegate cached in a static field.

Reply:
I thought the compiler would cache such a static local function delegate. I will save the delegate onto a static field to make this explicit.

Reply:
same concern here. w e own the entire stack.  i don't see the need for caches.  we can just expose an internal .Name property on DiagnosticAnalyzer if you need it right?

Reply:
I am going to rely on Sam's suggestion to rely on runtime caching these values and remove all the caching logic.